### ISSUE TITLE
Noisy inputs training

### ISSUE THREAD

**Issue by Howardyangyixuan** [czq142857/NMC#3]
Hi, zhiqin

I am trying to use the noisy inputs to train the NMC model from scratch to get similar results from paper but found the loss decrease far too slow compared to clean inputs.
for bool part:
[0/400] time: 1366  loss: 0.25179240  loss_bool: 0.91642386  loss_float: 0.00000000
[1/400] time: 2282  loss: 0.25031856  loss_bool: 0.91567051  loss_float: 0.00000000
[2/400] time: 3002  loss: 0.25059867  loss_bool: 0.91625684  loss_float: 0.00000000
[3/400] time: 3594  loss: 0.25166979  loss_bool: 0.91583282  loss_float: 0.00000000
[4/400] time: 4078  loss: 0.25226465  loss_bool: 0.91645384  loss_float: 0.00000000
...
[275/400] time: 45009  loss: 0.24075049  loss_bool: 0.91662174  loss_float: 0.00000000
[276/400] time: 45091  loss: 0.24094345  loss_bool: 0.91680372  loss_float: 0.00000000
[277/400] time: 45175  loss: 0.23898503  loss_bool: 0.91793585  loss_float: 0.00000000
[278/400] time: 45257  loss: 0.23988940  loss_bool: 0.91663897  loss_float: 0.00000000
[279/400] time: 45340  loss: 0.24025825  loss_bool: 0.91807079  loss_float: 0.00000000
 
Does it also occur to you when you train it? How many epochs did you train for both the bool and float part? and would you like to share noisy training weights with me to see more results from it? 
Great thanks!

**Comment by czq142857** [comment]
Hi,

During the noisy input experiment, we use [SIREN](https://github.com/vsitzmann/siren) to overfit the SDF of each shape, so the noise amount is actually small. According to your numbers, your input noise must be much larger. In that case, you could try to increase the receptive field of the network to resolve the ambiguity introduced by the noisy inputs, i.e., try using the network for voxels. Also, you may have noticed that if the inputs are SDF grids, the network will not predict the signs of the grid vertices, but use the signs of the input signed distances as the ground truth. This works well when the input is clean or has small noises, but may cause issues when the noise is larger. In that case, you may need to modify the code a bit to treat the input more like voxels.

Best,
Zhiqin

**Comment by Howardyangyixuan** [comment]
Got it. I will try it. Thanks for elaborating it for me and your consistent swift response!

Best Regards,
Yixuan
